### PR TITLE
Russian translation

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: ru/src/by-example.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# RTFM в примерах
+
+Эта часть книги представляет фреймворк Real Time For the Masses (RTFM)
+новым пользователям через примеры с растущей сложностью.
+
+Все примеры в этой книге можно найти в [репозитории] проекта на GitHub,
+и большинство примеров можно запустить на эмуляторе QEMU, поэтому никакого
+специальног оборудования не требуется to follow along.
```
Comment (little-arhat) [inline ru/src/by-example.md]: typo

Comment (little-arhat) [inline ru/src/by-example.md]: "to follow along" --> "чтобы их [примеры] выполнять"?

File: ru/src/by-example.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+[репозитории]: https://github.com/japaric/cortex-m-rtfm
+
+Чтобы запустить примеры на Вашем ноутбуке / PC, Вам нужна программа
+`qemu-system-arm`. Посмотрите [the embedded Rust book] на предмет инструкций
```
Comment (little-arhat) [inline ru/src/by-example.md]: "Посмотрите [the embedded Rust book] на предмет инструкций" => может лучше что-то типа "Вы можете найти инструкции в ..."?

File: ru/src/by-example/resources.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+Атрибут `app` имеет полную картину приложения, поэтому может оптимизировать доступ к
+`static`-переменным. В RTFM мы обращаемся к `static`-переменным, объявленным внутри
+псевдо-модуля `app` как к *ресурсам*. Чтобы получить доступ к ресурсу, контекст
+(`init`, `idle`, `interrupt` или `exception`) должен сначала определить
+аргумент `resources` в соответствующем атрибуте.
+
+В примере ниже два обработчика прерываний имеют доступ к одному и тому же ресурсу.
+Никакого `Mutex` в этом случае не требуется, потому что оба обработчика запускаются
+с обним приоритетом и никакого вытеснения быть не может.
```
Comment (little-arhat) [inline ru/src/by-example/resources.md]: "обним" -> "одним".

File: ru/src/by-example/singletons.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+когда-либо создан.
+
+[`owned-singleton`]: ../../api/owned_singleton/index.html
+[`Singleton`]: ../../api/owned_singleton_macros/attr.Singleton.html
+
+Заметьте, что когда Вы используете атрибут `Singleton`, Вым нужно иметь
+`owned_singleton` в зависимостях.
+
+Ниже, в примере, использован атрибут `Singleton` на куске памяти, а затем
+использован экземпляр одиночки как фиксированный по размеру пул памяти,
```
Comment (little-arhat) [inline ru/src/by-example/singletons.md]: Я бы изменил на "В примере ниже атрибутом `Singleton` аннотирован массив памяти, а экземпляр одиночки использован как фиксированный по размеру пул памяти..."

File: ru/src/by-example/tasks.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+{{#include ../../../ci/expected/message.run}}```
+
+## Ёмкость
+
+Диспетчеры задач *не* используют динамическое выделение памяти. Память
+необходимая для размещения сообщений, резервируется статически. Фреймворк
+зарезервирует достаточно памяти для каждого контекста, чтобы можно было вызвать
+каждую задачу как минимум единожды. Это разумно по умолчанию, но
+"внутреннюю" ёмкость каждой задачи можно контролировать используя аргумент
+`capacicy` атрибута `task`.
```
Comment (little-arhat) [inline ru/src/by-example/tasks.md]: "capacicy" -> "capacity"

File: ru/src/by-example/tips.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+# Советы и хитрости
+
+## Generics
+
+Ресурсы, совместно используемые двумя или более задачами реализуют трейт `Mutex`
```
Comment (little-arhat) [inline ru/src/by-example/tips.md]: нужна запятая после "задачами".

File: ru/src/by-example/tips.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+переписывания кода. Если Вы единообразно используете `lock`-и для доступа
+к данным в разделяемых ресурсах, тогда Ваш код продолжит компилироваться,
+когда Вы измените приоритет задач.
+
+## Запуск задач из RAM
+
+Главной целью переноса описания программы на RTFM в атрибуты в
+RTFM v0.4.x была возможность взаимодействия с другими атрибутами.
+Напримерe, атрибут `link_section` можно применять к задачам, чтобы разместить
+из в RAM; это может улучшить производительность в некоторых случаях.
```
Comment (little-arhat) [inline ru/src/by-example/tips.md]: "из" -> "их".

File: ru/src/by-example/types-send-sync.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+В примере ниже сгенерированы разные типы с помощью атрибута `app`.
+
+``` rust
+{{#include ../../../examples/types.rs}}
+```
+
+## `Send`
+
+[`Send`] - маркерный трейт для "типов, которые можно передавать через границы
```
Comment (little-arhat) [inline ru/src/by-example/types-send-sync.md]: кажется, в русскоязычных материалах про Rust слово "trait" переводят как "типаж", но лично мне не очень нравится %)

File: ru/src/by-example.md (hunk 1 above)
Comment (burrbull) [reply ru/src/by-example.md]: Спасибо, что просмотрели так оперативно. А то смотришь в книгу ...

File: ru/src/by-example/types-send-sync.md (hunk 8 above)
Comment (burrbull) [reply ru/src/by-example/types-send-sync.md]: Сталкивался как минимум с тремя разными переводами.

File: ru/src/README_RU.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
+# Real Time For the Masses
+
+Конкурентный фреймворк для создания систем реального времени.
+
+**ВАЖНО** Эта библиотека в настоящее время в состоянии пре-релиза (бета).
```
Comment (japaric) [inline ru/src/README_RU.md]: This note has been removed from the english version. Could you remove it here as well?

**Comment by japaric** [comment]
Looks good to me layout wise, but I can't read Russian ;-).

@little-arhat could I ask you to sign off this PR? You can find the bors documentation [here](https://bors.tech/documentation/).

bors delegate=little-arhat

**Comment by blandger** [comment]
@japaric did you see the discussion about translation main rust book here 
https://internals.rust-lang.org/t/better-support-translations-for-the-book/3003/24
? 
I've proposed kind of approach that makes less headache translate fast changing and updated docs.
I'm big fun of 'stamping' every paragraph before translating (or after if it's not late).

Currently there is no 'automatic' approach to put some ID+hash data into text. Probably it could be possible to do by processing current texts locally and updating English sources from time to time.

What do you think if I (for example) can put such 'meta-info' into English text and submit such PR into main repo? It would be really handy to track changes and translate by using 'timestamp meta info' provided in original text.

**Comment by burrbull** [comment]
Travis:
```
URL        `ru/index.html'
Name       `Russian'
Parent URL file:///tmp/tmp.htCSYJbCkp/book/index.html, line 144, col 43
Real URL   file:///tmp/tmp.htCSYJbCkp/book/ru/index.html
Check time 0.001 seconds
Result     Error: URLError: <urlopen error [Errno 2] No such file or directory: '/tmp/tmp.htCSYJbCkp/book/ru/index.html'>
```

**Comment by japaric** [comment]
@burrbull yes, my bad. More CI changes were needed. I have opened PR #139 with those changes.

@blandger The idea with the checksum sounds interesting but w/o support for automatically generating / updating them it sounds like it would be a non-trivial effort to maintain. Perhaps it would be simple to just add a "This translation corresponds to $COMMIT_HASH" at the beginning of each translation? Then the translator can do `git diff $COMMIT_HASH..HEAD book/en` to see the changes to the english version since the last update to the translation.

Closing this in favor of #139

**Pull request closed without merging**